.pipelines/npm/npm-cni-integration-test.yaml

```diff
@@ -0,0 +1,93 @@
+parameters:
+  clusterName: ""
+  os: ""
+  sub: ""
+  os_version: ""
+  tag: ""
+
+jobs:
+  - job: npm_k8se2e
+    displayName: "NPM k8s E2E"
+    dependsOn: ${{ parameters.dependsOn }}
+    condition: and(succeeded(), ${{ or(contains(parameters.os_version, '2022'), eq(parameters.os, 'linux')) }}  )
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+      demands:
+      - agent.os -equals Linux
+      - Role -equals Build
+    steps:
+      - task: AzureCLI@2
+        displayName: "Deploy NPM to Test Cluster"
+        inputs:
+          azureSubscription: ${{ parameters.sub }}
+          scriptType: "bash"
+          scriptLocation: "inlineScript"
+          inlineScript: |
+            set -ex
+
+            make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(make revision)
+
+            # deploy azure-npm
+            if ${{ lower(eq(parameters.os, 'windows')) }}
+            then
+              # Windows
+              kubectl apply -f https://raw.githubusercontent.com/Azure/azure-container-networking/master/npm/examples/windows/azure-npm.yaml
+              kubectl set image daemonset/azure-npm-win -n kube-system azure-npm=$IMAGE_REGISTRY/azure-npm:windows-amd64-ltsc2022-${{ parameters.tag }}
+              kubectl rollout status -n kube-system daemonset/azure-npm-win
+
+              # konnectivity agent tends to fail after rollout. Give it time to recover
+              sleep 60
+              # Taint Linux (system) nodes so windows tests do not run on them
+              kubectl taint nodes -l kubernetes.azure.com/mode=system node-role.kubernetes.io/control-plane:NoSchedule
+            else
+              # Linux
+              kubectl apply -f https://raw.githubusercontent.com/Azure/azure-container-networking/master/npm/azure-npm.yaml
+              kubectl set image daemonset/azure-npm -n kube-system azure-npm=$IMAGE_REGISTRY/azure-npm:${{ parameters.tag }}
+              kubectl rollout status -n kube-system daemonset/azure-npm
+            fi
+
+            kubectl get po -n kube-system -owide -A
+
+            # FQDN=`az aks show -n $CLUSTER_NAME -g $CLUSTER_NAME --query fqdn -o tsv`
+            FQDN=`az aks show -g ${{ parameters.clusterName }}-$(make revision) -n ${{ parameters.clusterName }}-$(make revision) --query fqdn -o tsv`
+            echo $FQDN
+            echo "##vso[task.setvariable variable=FQDN]$FQDN"
+
+      - download: current
+        artifact: Test
```

Comment:
If we are downloading can we name the artifact as npm_test or something. What does this produce as artifacts ?

Reply:
It was producing an artifact of k8s test suite fork. Saved roughly 8 minutes with this design. The name can be changed to npm_k8s. Will look like this when used `$(Pipeline.Workspace)/npm_k8s/e2e.test \...`